### DISCUSSION TITLE
Missing API Token

### DISCUSSION THREAD

**Discussion by jeenrish** [kirimwa-id/forum#119] [Tanya Jawab]
Halo, saya coba untuk pasang device dengan PHP, sudah berhasil sekali kemudian saya tidak bisa melakukan scan qr code (qr-nya tidak muncul). Setelah itu saya coba ulang step dari awal, dan saya mendapat pesan Missing API Token seperti ini. Saya sangat awam dengan bahasa programming seperti ini, mohon bantuannya..

> Exception Object ( [message:protected] => {"message":"Missing API token."} [string:Exception:private] => [code:protected] => 401 [file:protected] => C:\xampp\htdocs\php-api-kirimwa\functions.php [line:protected] => 47 [trace:Exception:private] => Array ( [0] => Array ( [file] => C:\xampp\htdocs\php-api-kirimwa\add-device.php [line] => 15 [function] => apiKirimWaRequest [args] => Array ( [0] => Array ( [token] => [url] => https://api.kirimwa.id/v1/devices [method] => POST [payload] => {"device_id":false} ) ) ) ) [previous:Exception:private] => )

**Comment by kafaa-dev** [comment]
Tokennya belum dikirimkan gan

**Reply by jeenrish** [reply]
halo, maksudnya belum dikirimkan gimana yaa kak? soalnya token API-nya sudah saya ganti sesuai dengan yang di email, tapi pas diload hasilnya tetep gitu

**Reply by rioastamal** [reply]
@jeenrish silahkan post kodenya disini dengan menghilangkan informasi yang sensitif misal token atau no. HP

Cek disini untuk cara mengajukan pertanyaan:
https://github.com/kirimwa-id/forum/discussions/2

**Reply by papabreng** [reply]
> @jeenrish silahkan post kodenya disini dengan menghilangkan informasi yang sensitif misal token atau no. HP
> 
> Cek disini untuk cara mengajukan pertanyaan: #2

saya mengalami hal yang sama dengan kode seperti ini, mohon bantuannya

<?php
require './functions.php';


// Lihat apiKirimWaRequest() pada Contoh Integrasi diatas
try {
  $reqParams = [
    'token' => getenv('xxxxx'),
    'url' => 'https://api.kirimwa.id/v1/devices'
  ];

  $response = apiKirimWaRequest($reqParams);
  echo $response['body'];
} 
  catch (Exception $e) {
  print_r($e);
}
